diff --git a/dlls/winex11.drv/keyboard.c b/dlls/winex11.drv/keyboard.c
index 3907828..98d5aa5 100644
--- a/dlls/winex11.drv/keyboard.c
+++ b/dlls/winex11.drv/keyboard.c
@@ -1141,7 +1141,11 @@ static void X11DRV_send_keyboard_input( HWND hwnd, WORD vkey, WORD scan, UINT fl
     input.ki.time        = time;
     input.ki.dwExtraInfo = 0;
 
+#ifdef __ANDROID__
+    NtUserSendHardwareInput( hwnd, SEND_HWMSG_NO_RAW, &input, 0 );
+#else
     NtUserSendHardwareInput( hwnd, 0, &input, 0 );
+#endif
 }
 
 
